core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1567,24 +1518,104 @@ class KafkaApis(val requestChannel: RequestChannel,
     })
 
     if(shareSessionEpoch == ShareFetchMetadata.FINAL_EPOCH && !shareFetchDataWithPartitionMaxBytes.isEmpty) {
+      isValidRequest = false
```

Comment:
You don't need this variable. You can just use an `} else {` instead of the `if(isValidRequest)` on line 1532.